### PR TITLE
`Assessment`: Show average rating for exercises in course assessment dashboard

### PR DESCRIPTION
**Pull request by Olimbia23**
#### General

- [X] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] * This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ] * I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] * I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [X] I implemented the changes with a good performance and prevented too many database calls.
- [X] I documented the Java code using JavaDoc style.
#### Client
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [ ] * I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ] * I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [X] I documented the TypeScript code using JSDoc style.
- [X] I added multiple screenshots/screencasts of my UI changes.
- [X] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] * I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] * I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

[ ]* - not needed

### Motivation and Context

It would beneficial for the instructors as well tutors to see the average rating of the exercise in the course.

### Description

New column in the exercise table is introduced as well as the necessary data is computed on the server and passed to the client in `for-assessment-dashboard` call.
The values in the column are `n.a.` in cases when no ratings are provided. The same logic is adjusted in the tutor-leaderboard for average score as well as average rating columns.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis (either as instructor or tutor)
2. Navigate to Course Administration
3. Navigate to Assessment Dashboard
4. You should be able to see a new column in the exercise table named **Average Rating**

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

no change

#### Before
Screenshot 2021-11-08 at 8 37 57 PM

-------------------------------------------

### CODE REVIEW

**Review by Hialus — APPROVED** [review]
Tested on TS1.
Tested for exercises with and without ratings. Both cases work as they should, and the average score is calculated correctly. The sorting of the column also works reasonably.
The code changes also look good.
Well done @Olimbia23 👍🏻

**Review by imstreetspirit — APPROVED** [review]
tested on ts1, works as expected. code changes look good to me 👍

**Review by MerlinMehmed — APPROVED** [review]
Tested on TS1, works as expected. Code changes also look good to me. Good job @Olimbia23

File: src/main/java/de/tum/in/www1/artemis/domain/assessment/dashboard/ExerciseRatingCount.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package de.tum.in.www1.artemis.domain.assessment.dashboard;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record ExerciseRatingCount(Double averageRating, Long numberOfRatings) {
+}
```
Comment (lschlesinger) [inline src/main/java/de/tum/in/www1/artemis/domain/assessment/dashboard/ExerciseRatingCount.java:7]: you could add a doc here as we do for new classes

**Review by lschlesinger — APPROVED** [review]
Code looks good. I think you could add a short documentation of the newly introduced record.
I did not know that one could rate exercises :) the average rating is valuable information for instructors and tutors 👍

File: src/main/java/de/tum/in/www1/artemis/domain/assessment/dashboard/ExerciseRatingCount.java (hunk 1 above)
Comment (Olimbia23) [reply src/main/java/de/tum/in/www1/artemis/domain/assessment/dashboard/ExerciseRatingCount.java:7]: I can add this documentation in another PR 👍

**Pull request merged**